### PR TITLE
Added SetUserData to run Unit Test of Post Policy

### PR DESCRIPTION
**Pull request by sinhaashish**
To execute the test cases for [PR](https://github.com/minio/minio/pull/6628) added a function `SetUserData` . This will resolve the [issue ](https://github.com/minio/minio/issues/6559)

### CODE REVIEW

File: post-policy.go [hunk 1]
```diff
@@ -206,6 +206,28 @@ func (p *PostPolicy) SetUserMetadata(key string, value string) error {
 	return nil
 }
 
+// SetUserData - Set user data as a key/value couple.
+// Can be retrieved through a HEAD request or an event.
+func (p *PostPolicy) SetUserData(key string, value string) error {
+	if strings.TrimSpace(key) == "" || key == "" {
```
Comment (kannappanr) [inline post-policy.go]: If `key==""` is `true` then `strings.TrimSpace(key) == ""` will also be `true`

File: post-policy.go [hunk 2]
```diff
@@ -206,6 +206,28 @@ func (p *PostPolicy) SetUserMetadata(key string, value string) error {
...
+		return ErrInvalidArgument("Key is empty")
+	}
+	if strings.TrimSpace(value) == "" || value == "" {
+		return ErrInvalidArgument("Value is empty")
+	}
+	headerName := fmt.Sprintf("x-amz-%s", key)
```
Comment (Praveenrajmani) [inline post-policy.go]: Can we have canonical header prefix here ?

**Comment by sinhaashish** [comment]
>Looks like we already have SetUserMetadata https://github.com/minio/minio-go/blob/774475480ffe1bffa9dae62ea98f6ea75e8cf567/post-policy.go#L189. We can use this instead.
`SetUserMetadata` contained headerName as `x-amz-meta` but the ask was for `x-amz` . so added this function.

File: post-policy.go (hunk 2 above)
Comment (sinhaashish) [reply post-policy.go]: Canonical header is used in minio/minio .. The test case has it small.

File: post-policy.go (hunk 1 above)
Comment (sinhaashish) [reply post-policy.go]: removed `strings.TrimSpace(key) == ""`

**Pull request merged**